src/main/java/io/cdap/plugin/hive/action/common/HiveConfig.java

```diff
@@ -32,35 +34,147 @@
+  private String connectionString;
 
   @Name(USER)
   @Description("User to use to connect to hive metastore database. " +
     "Optional for databases that do not require authentication.")
   @Nullable
   @Macro
-  public String user;
+  private String user;
 
   @Name(PASSWORD)
   @Description("Password to use to connect to hive metastore database. " +
     "Optional for databases that do not require authentication.")
   @Nullable
   @Macro
-  public String password;
+  private String password;
 
   @Name(STATEMENT)
   @Description("Hive command to execute")
   @Macro
-  public String statement;
+  private String statement;
 
   public HiveConfig(String connectionString, String user, String password, String statement) {
     this.connectionString = connectionString;
     this.user = user;
     this.password = password;
     this.statement = statement;
   }
+
+  private HiveConfig(Builder<?> builder) {
+    connectionString = builder.connectionString;
+    user = builder.user;
+    password = builder.password;
+    statement = builder.statement;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static Builder builder(HiveConfig copy) {
+    return builder()
+      .setConnectionString(copy.connectionString)
+      .setUser(copy.user)
+      .setPassword(copy.password)
+      .setStatement(copy.statement);
+  }
+
+  public String getConnectionString() {
+    return connectionString;
+  }
+
+  @Nullable
+  public String getUser() {
+    return user;
+  }
+
+  @Nullable
+  public String getPassword() {
+    return password;
+  }
+
+  public String getStatement() {
+    return statement;
+  }
+
+  public void validate(FailureCollector failureCollector) {
+    if (!containsMacro(CONNECTION_STRING) && !connectionString.startsWith(CONNECTION_STRING_PREFIX)) {
+      failureCollector.addFailure(
+        "Invalid connection string.",
+        "Connection String must comply with format - jdbc:vertica://<VerticaHost>:<portNumber>/<databaseName>")
+        .withConfigProperty(CONNECTION_STRING);
+    }
+
+    if (!containsMacro(USER) && !containsMacro(PASSWORD)) {
+      if (Strings.isNullOrEmpty(user) && !Strings.isNullOrEmpty(password)) {
+        failureCollector.addFailure(
+          "Username is not specified.",
+          "Ensure both username and password are provided.")
+          .withConfigProperty(USER).withConfigProperty(PASSWORD);
+      }
+      if (!Strings.isNullOrEmpty(user) && Strings.isNullOrEmpty(password)) {
+        failureCollector.addFailure(
+          "Password is not specified.",
+          "Ensure both username and password are provided.")
+          .withConfigProperty(USER).withConfigProperty(PASSWORD);
+      }
+    }
+  }
+
+  public void validateImportStatment(FailureCollector failureCollector) {
+    if (!containsMacro(STATEMENT)) {
+      // Load command should not allow local storage
+      if (statement.length() >= 15 && statement.substring(10, 15).equalsIgnoreCase("LOCAL")) {
```

Comment:
10 to 15? Can't the statement have extra space or something?

Also please don't use magic numbers should constants.

Reply:
This is the based implimentation.
In information about LOAD statement I notfound any information about extra spaces.
LOAD statement has next syntax:
LOAD DATA [LOCAL] INPATH 'filepath' [OVERWRITE] INTO TABLE tablename [PARTITION (partcol1=val1, partcol2=val2 ...)]
Currently I updated checking forthis case.